indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -387,6 +392,25 @@ sealed class Asn1Element(
     }
 }
 
+/**
+ * asserts that this element's tag matches [tag].
+ *
+ * @throws Asn1TagMismatchException on failure
+ */
+@Throws(Asn1TagMismatchException::class)
+inline fun <reified T : Asn1Element> T.assertTag(tag: Asn1Element.Tag): T {
+    if (this.tag != tag) throw Asn1TagMismatchException(tag, this.tag)
+    return this
+}
+
+/**
+ * Asserts only the tag number, but neither class, nor CONSTRUCTED bit.
+ * @see assertTag
+ * @throws Asn1TagMismatchException on failure
+ */
+@Throws(Asn1TagMismatchException::class)
+inline fun <reified T : Asn1Element> T.assertTag(tagNumber: ULong): T = assertTag(tag.withNumber(tagNumber))
```

Comment:
This trickery where we construct a fake tag to assert against feels like an ugly workaround for `Asn1TagMismatchException` needing tags

Reply:
I'm open for ideas, but the mismatch exception should keep both parameters

Reply:
hm, yeah, then i don't know either